topsf/tasks/inference_v2/multi_dim_fit.py

```diff
@@ -0,0 +1,154 @@
+        description="Save the workspace",
+    )
+
+    snapshot_name = luigi.Parameter(
+        default="",
+        significant=False,
+        description="Name of the snapshot",
+    )
+
+    freeze_fit_parameters = law.CSVParameter(
+        significant=False,
+        description="Freeze parameters for the fit",
+    )
+
+    # upstream requirements
+    reqs = Requirements(
+        RemoteWorkflow.reqs,
+        CreateDatacards=CreateDatacards,
+        CreateWorkspace=CreateWorkspaceV2,
+        GenToys=GenToysV2,
+    )
+
+    def workflow_requires(self):
+        reqs = super().workflow_requires()
+        reqs["workspace"] = self.reqs.CreateWorkspace.req(self)
+        if self.mode_inst == "exp":
+            reqs["toy_file"] = self.reqs.GenToys.req(self)
+        return reqs
+
+    def requires(self):
+        reqs = {
+            "workspace": self.reqs.CreateWorkspace.req(self),
+        }
+        if self.mode_inst == "exp":
+            reqs["toy_file"] = self.reqs.GenToys.req(self)
+        return reqs
+
+    @property
+    def mdf_name(self):
+        name = f"mdf__{self.mode_inst}"
+        return name
+
+    def store_parts(self) -> law.util.InsertableDict:
+        parts = super().store_parts()
+        parts.insert_after("fit_v2", "gen_toys", self.mdf_name)
+        return parts
+
+    def output(self):
+        output_dict = {
+            f"mdf_{self.mode_inst}_file": self.target(
+                f"higgsCombine_{self.mode_inst}.MultiDimFit.mH120.root",
+            ),
+            f"mdf_{self.mode_inst}_frozen_file": self.target(
+                f"higgsCombine_{self.mode_inst}_frozen.MultiDimFit.mH120.root",
+            ),
+            f"fit_{self.mode_inst}_result": self.target(
+                f"multidimfit_{self.mode_inst}.root",
+            ),
+            f"fit_{self.mode_inst}_frozen_result": self.target(
+                f"multidimfit_{self.mode_inst}_frozen.root",
+            ),
+            f"mdf_{self.mode_inst}_log": self.target(
+                f"mdf_{self.mode_inst}.log",
+            ),
+            f"mdf_{self.mode_inst}_frozen_log": self.target(
+                f"mdf_{self.mode_inst}_frozen.log",
+            ),
+        }
+        return output_dict
+
+    @law.decorator.log
+    @law.decorator.safe_output
+    def run(self):
+        # if self.mode_inst == "exp":
+        self.combine_method_inst = "MultiDimFit"
+        input_workspace = self.input()["workspace"]["workspace"].path
+        frozen_sys_workspace = self.output()[f"mdf_{self.mode_inst}_file"].path
+        if self.mode_inst == "exp":
+            toy_file = self.input()["toy_file"]["toy_file"].path
+        output_mdf_file = self.output()[f"mdf_{self.mode_inst}_file"].path
+        output_dirname = os.path.dirname(output_mdf_file) + "/"
+        print(f"output_dirname: {output_dirname}")
+        # touch output_dirname
+        if not os.path.exists(output_dirname):
+            os.makedirs(output_dirname)
```

Comment:
There is a shorter idiom for this ;)
```suggestion
        os.makedirs(output_dirname, exist_ok=True)
```